doc/specs/#1571 - New Tab Menu Customization/#1571 - New Tab Menu Customization.md

```diff
@@ -0,0 +1,125 @@
+---
+author: Mike Griese @zadjii-msft
+created on: 2020-5-13
+last updated: 2020-5-13
+issue id: 1571
+---
+
+# New Tab Menu Customization
+
+## Abstract
+
+Many users have lots and _lots_ of profiles that they use. Some of these
+profiles the user might not use that frequently. When that happens, the new tab
+dropdown can become quite cluttered.
+
+A common ask is for the ability to reorder and reorganize this dropdown. This
+spec provides a design for how the user might be able to specify the
+customization in their settings.
+
+## Inspiration
+
+Largely, this spec was inspired by discussion in
+[#1571](https://github.com/microsoft/terminal/issues/1571#issuecomment-519504048)
+and the _many_ linked threads.
+
+## Solution Design
+
+This design proposes adding a new setting `"newTabMenu"`. When unset, (the
+default), the new tab menu is populated with all the profiles, in the order they
+appear in the users settings file. When set, this enables the user to control
+the appearance of the new tab dropdown. Let's take a look at an example:
+
+```json
+{
+    "profiles":{ ... },
+    "newTabMenu": [
+        { "type":"profile", "profile": "cmd" },
+        { "type":"profile", "profile": "Windows PowerShell" },
+        { "type":"separator" },
+        {
+            "type":"folder",
+            "name": "ssh",
+            "entries":[
+                { "type":"profile", "profile": "Host 1" },
+                { "type":"profile", "profile": "8.8.8.8" },
+                { "type":"profile", "profile": "Host 2" }
+            ]
+        },
+        { "type":"separator" },
+        { "type":"profile", "profile": "Ubuntu-18.04" },
+        { "type":"profile", "profile": "Fedora" }
+    ]
+}
+```
+
+If a user were to use this as their new tab menu, that they would get is a menu
+that looks like this:
+
+![fig 1](Menu-Customization-000.png)
+
+_fig 1_: A _very rough_ mockup of what this feature might look like
+
+There are three `type`s of objects in this menu:
+* `"type":"profile"`: This is a profile. Clicking on this entry will open a new
+  tab, with that profile. The profile is identified with the `"profile"`
+  parameter, which accepts either a profile `name` or GUID.
+* `"type":"separator"`: This represents a XAML `MenuFlyoutSeparator`, enabling
+  the user to visually space out entries.
+* `"type":"entries"`: This represents a nested menu of entries.
+  - The `"name"` property provides a string of text to display for the group.
+  - The `"entries"` property specifies a list of menu entries that will appear
+    nested under this entry. This can contain other `"type":"entries"` groups as
+    well!
+

```

Comment:
For the "folder" type, I think we're also missing the icon. It would be cool to link an icon like we can with all of the other profiles.

That way, I could create a folder for all my WSL distros, and the folder `MenuFlyoutItem` would have a WSL icon on it.